plugins/org.eclipse.fordiac.ide.model.search/src/org/eclipse/fordiac/ide/model/search/ModelSearchQuery.java

```diff
@@ -297,23 +301,57 @@ private boolean matchEObject(final INamedElement modelElement, final IProgressMo
 						|| (varDecl.getType() != null && varDecl.getType().getTypeEntry() != null
 								&& compareStrings(varDecl.getType().getTypeEntry().getFullTypeName()));
 			}
-			if (modelElement instanceof Algorithm || modelElement instanceof Method) {
-				for (final String fqn : VariableOperations.getAllDependencies(modelElement)) {
-					if (compareStrings(fqn)) {
-						return true;
-					}
+			if ((modelElement instanceof Algorithm || modelElement instanceof Method)
+					&& matchInAlgOrMethod(modelElement)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	private boolean matchInAlgOrMethod(final INamedElement modelElement) {
+		final String text = getImplText(modelElement);
+		boolean preCheck = false;
+
+		if (text != null) {
+			String searchStringLC = modelQuerySpec.getSearchString().toLowerCase();
+			final int lastPackageSepeartor = searchStringLC.lastIndexOf("::"); //$NON-NLS-1$
+			if (lastPackageSepeartor != -1) {
+				searchStringLC = searchStringLC.substring(lastPackageSepeartor + 2);
+			}
+			if (text.toLowerCase().contains(searchStringLC)) {
```

Comment:
What if the search string contains wildcards? If we search using a case-insensitive Pattern, we could also avoid converting to lower case.

Reply:
That is a good idea. I was also not happy about the lower case conversion.